datanode/sqlstore/markets.go

```diff
@@ -233,3 +237,41 @@ func (m *Markets) GetAllPaged(ctx context.Context, marketID string, pagination e
 	m.allCache[key] = cacheValue{markets: markets, pageInfo: pageInfo}
 	return markets, pageInfo, nil
 }
+
+func (m *Markets) ListSuccessorMarkets(ctx context.Context, marketID string, fullHistory bool) ([]entities.Market, error) {
+	if marketID == "" {
+		return nil, errors.New("invalid market ID. Market ID cannot be empty")
+	}
+
+	args := make([]interface{}, 0)
+
+	lineageFilter := ""
+
+	if !fullHistory {
+		lineageFilter = "and vega_time >= (select vega_time from lineage_root)"
+	}
+
+	query := fmt.Sprintf(`
+		with lineage_root(root_id, vega_time) as (
+            select root_id, vega_time
+            from market_lineage
+            where market_id = %s
+        ), lineage(market_id, parent_id, root_id) as (
```

Comment:
I don't really see something here that creates a market history in a sort of doubly-linked list way. On this side of things (ie datanode) markets could (And should) have the ID of their parent and the market that succeeded them.

Reply:
The `market_lineage` table has a mapping for the market_id, it's parent_market_id, and the `root_id` which is the basically the first market in the succession line, I guess a better name for it could be lineage_id, but basically you can get the root_id for any market, and from that get the full succession line, or you can use the parent_market_id to get the child market_id, or vice versa.

```sql
CREATE TABLE IF NOT EXISTS market_lineage (
    market_id BYTEA NOT NULL PRIMARY KEY,
    parent_market_id BYTEA, -- the root market in the lineage chain will not have a parent, but all subsequent children will
    root_id BYTEA NOT NULL, -- market id of the first market in the lineage chain
    vega_time TIMESTAMP WITH TIME ZONE NOT NULL
);
```

Reply:
OK, so the comment doesn't apply here, but I think we should return markets with a successor_id and parent_id field, in case the market was requested without the full history, we still would want to be able to fetch the previous/next market in line easily

Reply:
The parent_market_id is already a member of the market objects that are returned, but you're right, it would be useful to include the successor too.